books/book.js

```diff
@@ -30,10 +30,75 @@ function displayBook(book) {
   const buttonBasket = `<button class = 'basket' > Add to basket </button>`;
   const buttonWish = `<button class = 'wishlist' > Add to wishlist </button>`;
+  const loginMsg = `<a class = 'login-msg' href = '../login/login.html'> Please Login To Buy Books </a>`;
   const bookInfoDiv = `<div class = 'book-info'> ${title} ${author} ${bookYear} ${topic} ${synopsis} </div>`;
   const bookCoverDiv = `<div class = 'book-cover-div'><img class = 'book-cover' src = ${book.cover_url}></div>`;
-  const bookPurchase = `<div class = 'book-purchase'> ${bookPrice} ${bookPurchaseH3} ${bookPurchaseP} ${buttonBuy} ${buttonBasket} ${buttonWish}</div>`;
+  const bookPurchase = `<div class = 'book-purchase'> ${bookPrice} ${bookPurchaseH3} ${bookPurchaseP} ${buttonBuy} ${buttonBasket} ${buttonWish} ${loginMsg}  </div>`;
   currentBook = `<div class = 'book'>${bookInfoDiv} ${bookCoverDiv}  ${bookPurchase}</div>`;
 
   main.innerHTML = currentBook;
+  let msg = document.querySelector(".login-msg");
+  let buttonsBuy = document.querySelectorAll(".book button");
+  let usernameSessionValue = sessionStorage.getItem("username");
+  let loginBtn = document.querySelector(".login-btn");
+  let username = document.querySelector(".username");
+  let signOutBtn = document.querySelector(".sign-out");
+  if (usernameSessionValue) {
+    buttonsBuy.forEach((btn) => {
+      btn.style.display = "block";
+      username.style.display = "block";
+      username.innerHTML = usernameSessionValue + '<i class="fas fa-user"></i>';
+      loginBtn.style.display = "none";
+      signOutBtn.style.display = "block";
+    });
+    msg.style.display = "none";
+  } else {
+    buttonsBuy.forEach((btn) => {
+      btn.style.display = "none";
+    });
+  }
+
+  signOutBtn.addEventListener("click", signOut);
+  function signOut() {
+    window.open("../index.html", "_self");
+    sessionStorage.removeItem("username");
+    sessionStorage.removeItem("password");
+  }
+  let buyButton = document.querySelector(".buy");
+  buyButton.addEventListener("click", purchaseBook);
+  function purchaseBook() {
+    // Creating Date object and formatting in a way that it has ISO format and that it does not include hours , minutes and seconds
+    let currentDate = new Date();
+    let formattedDate = `${currentDate.getFullYear()}-${
+      currentDate.getMonth() + 1
+    }-${currentDate.getDate()}`;
+    // Getting username and password from session storage
+    const usernameSessionValue = sessionStorage.getItem("username");
+    const passwordSessionValue = sessionStorage.getItem("password");
+    // Authentication part //
+    const base64EncodedString = btoa(
+      `${usernameSessionValue}:${passwordSessionValue}`
+    );
+    fetch("https://bookshop-api.mirkwood.dev/purchases/", {
+      method: "POST",
+      headers: {
+        Authorization: `Basic  ${base64EncodedString}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        id: 7,
```

Comment:
Id is generated by the backend, you do not need to specify it here

Reply:
Alright